### PR TITLE
Add static website hosting to buckets

### PR DESCRIPTION
**Pull request by giorgiosironi**
To turn this on, we need a WebsiteConfiguration on the Bucket plus a related BucketPolicy to open the contents to the public

### CODE REVIEW

**Comment by giorgiosironi** [comment]
Just used this to create `error-pages--end2end` (and soon `error-pages--prod`).

File: src/buildercore/trop.py [hunk 1]
```diff
@@ -360,6 +360,16 @@ def render_s3(context, template):
             'DeletionPolicy': context['s3'][bucket_name]['deletion-policy'].capitalize()
         }
         bucket_title = _sanitize_title(bucket_name) + "Bucket"
+        if context['s3'][bucket_name]['cors']:
+            props['CorsConfiguration'] = s3.CorsConfiguration(
+                CorsRules=[
+                    s3.CorsRules(
+                        AllowedHeaders=['*'],
+                        AllowedMethods=['GET', 'HEAD'],
+                        AllowedOrigins=['*']
```
Comment (lsh-0) [inline src/buildercore/trop.py]: this should be enough to satisfy @davidcmoulton requirements, cheers

Comment (giorgiosironi) [inline src/buildercore/trop.py]: it seems standard, I don't know who would be crazy enough to allow writes to a bucket from a page loaded in a website

Comment (lsh-0) [inline src/buildercore/trop.py]: maybe if it were within an intranet or secured or something. I like the idea of POSTing something to a bucket and then having the change to a bucket's contents trigger other changes, reactive style

Comment (giorgiosironi) [inline src/buildercore/trop.py]: that's called `elife-bot` and you rightly often elaborated on its level of internal quality :)

Comment (lsh-0) [inline src/buildercore/trop.py]: yeah, but the bot is a sprawling stateful mess. I'd have all of the state contained within the buckets and everything else be a dumb almost-pure function.

**Comment by lsh-0** [comment]
jah, merge man.

**Pull request merged**